OptimizelySDK/Odp/OdpEventManager.cs

```diff
@@ -349,7 +354,17 @@ public void IdentifyUser(string userId)
         /// <param name="odpConfig">Configuration object containing new values</param>
         public void UpdateSettings(OdpConfig odpConfig)
         {
+            if (odpConfig == null)
+            {
+                return;
+            }
+            Flush();
```

Comment:
why this `Flush` is called?

Reply:
We want to flush any events currently in the queue using the previous config then apply the new config.  

I've added to the method doc